### ISSUE TITLE
about changes for non-ssl RP and options {privacyPolicy,termsOfService}

### ISSUE THREAD

**Issue by jrgm** [mozilla/persona#3469]
So I see the commit for this change, https://github.com/mozilla/browserid/commit/5c81116a229d0776dfc6ddf2af8b5ecc5305048b, and I know this was done for b2g reasons.

At any rate, late as usual, but wondering about existing RP on desktop, where we are disabling them from using these features if they don't use https.

This may be a necessary regression, but at least the documentation should be updated if this is not a permanent change in the api:

```
https://developer.mozilla.org/en-US/docs/Web/API/navigator.id.request
https://developer.mozilla.org/en-US/docs/Web/API/navigator.id.get
```

Related: clicking on the tos and pp links in the dialog with desktop login.persona.org will now go to /embedded_tos and /embedded_privacy instead of the previous login.persona.org/{tos,privacy} url.

**Comment by shane-tomlinson** [comment]
> Related: clicking on the tos and pp links in the dialog with desktop login.persona.org will now go to /embedded_tos and /embedded_privacy instead of the previous login.persona.org/{tos,privacy} url.

We made this change so that headers and footers would not be shown in the embedded context, knowing it affected the old-style open in new tab as well. We decided we would revisit this at a later time.

**Comment by shane-tomlinson** [comment]
@callahad - a bit of context for you:

Native environments such as FirefoxOS do not support window.open and cannot open new tabs in which the TOS/PP agreements can be displayed. To get around this, we decided to show the TOS/PP agreements in a sandboxed [http://www.html5rocks.com/en/tutorials/security/sandboxed-iframes/], embedded IFRAME. The IFRAME does not allow Javascript to run. To avoid mixed mode (http iframe embedded in an https dialog) messages from the browsers, we decided to limit TOS/PP agreements to RPs that serve over https.

TOS/PP agreements can be quite long, and any IFRAME must have the ability to scroll. Desktop browsers all act sanely, mobile browsers do not. The Android native browser acts differently to Firefox mobile and Mobile Safari. The CSS that allows Android to scroll breaks the other two. The CSS that fixes the FF/Mobile Safari breaks Android. The CSS must be done inside of the IFRAME. Every site that serves a TOS/PP agreement would have to deal with this. This sucks.

So, we decided - why not open the TOS/PP agreement as we used to _for all browsers that support it_, and fall back to the IFRAME for everybody else. So, we initially try window.open. If that doesn't work, we then load up the TOS/PP agreement in an IFRAME. So far, we know of only FirefoxOS that requires the IFRAME, but I have not tried the Android and iOS SDKs. FirefoxOS acts sanely with how it scrolls content within the IFRAME and everybody is happy - except for RPs that are served using http.
